### DISCUSSION TITLE
Pardot Custom Mapping not returning field data

### DISCUSSION THREAD

**Discussion by ch-turnstyle** [solspace/craft-freeform#1777] [Questions & Answers]
### What happened?

I've successfully authenticated with Pardot for my forms, and am able to map to the Pardot Prospect fields. However, the Custom Mapping fields do not come back with any data, even though my client has indicated they exist. There are no errors, just a message that says no data present.

All the custom field types I wish to map to are listed in the types within Freeform's `getCustomFields` function, so I don't think it's a field type issue.

### How can we reproduce this?

1. Integrate and authenticate Pardot.
2. Create a new form, toggle on Map to Custom via the Integrations tab.
3. No data present message.

### Freeform Edition

Pro

### Freeform Version

5.8.4

### Craft Version

Craft Pro 5.5.7

### When did this issue start?

Fresh install of Freeform

**Comment by kjmartens** [comment]
Hi @ch-turnstyle,

I'm very sorry for the trouble and the delay...

I have a few questions:

1. Does clicking the "refresh" icon button in the form builder (by the field mapping table) resolve anything?
2. Are there any errors in the browser console in the form builder, and can you [check the Network Response](https://docs.solspace.com/craft/freeform/v5/guides/testing/troubleshooting-form-issues/#check-the-browser-console-and-network-response)?
3. Is there a chance that the connection to Pardot (when setting up initially) didn't go through?
4. What happens if you try resaving the Pardot integration inside the Freeform settings area? Does it present you with OAuth permission screen, etc?
5. Can you include a screenshot of your Freeform Diagnostics page? 🙂

**Comment by ch-turnstyle** [comment]
Hi @kjmartens, thanks for getting back to me!

1. I've clicked the refresh a few times, but still just an empty array coming back in the Network panel.
2. There are no errors coming back from the request - just a successful GET 200, with a response of an empty array.
3. The connection is successful, as we were able to submit a form to their Pardot endpoint with data being populated for the Prospect mapped fields.
4. Only the client has a login to the Pardot/Salesforce side of things, and we had to set up the integration a few times as we had the wrong Business Unit ID/permissions at first. Each time we tried the OAuth permission screen did come up, and eventually we got the correct Business Unit ID/api permissions, which is currently saved and working for the Prospect mapping. I also confirmed with the client that they only have one Pardot integration.
5. Screenshot below, and note that we had to enable Post Forwarding that forwards to a custom form handler that they created. Ideally of course, we'd like to just map the fields. Also, i'm importing the scripts in my template using
```
{{ freeform.loadScripts([
  "freeform",
  "recaptcha.v3",
]) }}
```
![Screenshot 2025-01-13 at 11 40 22 AM](https://github.com/user-attachments/assets/01a43794-6b2e-4db5-8a0c-859c5707edf1)

**Comment by kjmartens** [comment]
Thank you for all of the additional details @ch-turnstyle.

I am wondering if the initial integration in Freeform settings somehow got corrupted. Is it possible to delete it entirely (the profile from the settings area), open and resave the form inside the form builder, and then create the integration again and see if that resolves the issue?

If not, are you able to share your database with us for review? If you could send that to us at support@solspace.com, I could have a developer check into it. 🙂

**Reply by ch-turnstyle** [reply]
Hi @kjmartens, apologies for the delay --this is a client project, and they haven't decided on the steps they'd like to take here, so I'm unable to try either option just yet. Thank you for converting to a discussion, and I'll be sure to reach out if/when they decide what they'd like to do.
